### DISCUSSION TITLE
When adding standard CORS headers, it fails cause they are duplicated only for OPTIONS method

### DISCUSSION THREAD

**Discussion by Imanflow** [fastapi/fastapi#11851] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Any

from fastapi import FastAPI
from fastapi.openapi.docs import get_swagger_ui_html
from fastapi.openapi.utils import get_openapi
from pydantic import BaseModel
from starlette.middleware.cors import CORSMiddleware
from starlette.responses import JSONResponse

from processing import compute_intersections


class Geom(BaseModel):
    type: str
    coordinates: Any


class Request(BaseModel):
    geometry: Any
    storm_surge: str
    confidence: str
    ssp: str
    year: int


app = FastAPI()

origins = ["*"]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=False,
    allow_methods=["*"],
    allow_headers=["*"],
)

@app.post("/areas", include_in_schema=True)
def compute_afected_areas(data: Request):

    results = True

    return results


@app.options("/areas", include_in_schema=False)
def compute_afected_areas():


    return True
```

### Description

If I don't add code code for CORS middleware and I use a CORS tester, i get the following results:

For POST method it tells me it fails cause CORS is not set up (expected), but for OPTIONS method it tells me it works, with this answer:

```
access-control-allow-origin: *
cf-cache-status: DYNAMIC
cf-ray: 8a42dea9a4b5384c-MAD
connection: keep-alive
content-length: 4
content-type: application/json
date: Tue, 16 Jul 2024 15:01:49 GMT
server: cloudflare
```

The problem is that if I set up the CORS, then it says that it works for POST, but not for OPTIONS since the header is set up twice:

`access-control-allow-origin: *, *
`

I checked and there is no duplication in my nginx webserver. Where the hell can that configuration be coming from, and only for the OPTIONS method, and no others?

### Operating System

Linux

### Operating System Details

Ubuntu 22.04.4 LTS

### FastAPI Version

0.100.0

### Pydantic Version

2.0.2

### Python Version

3.10.12

**Comment by AIdjis** [comment]
Hello, try to set the allow_credentials to True like this :
```python
app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)
```

**Reply by Imanflow** [reply]
Before I had in on True, until i found this: https://github.com/tiangolo/fastapi/issues/5263#issuecomment-1211997428

**Reply by AIdjis** [reply]
i just read this it is form fastapi documentation : 
allow_credentials - Indicate that cookies should be supported for cross-origin requests. Defaults to False. Also, allow_origins cannot be set to  ["*"]  for credentials to be allowed, origins must be specified. 

so In production, if allow_credentials is set to True, you cannot set allowed_origins to  ["*"] (which would allow all origins). Instead, you must explicitly specify the origins that are permitted. Allowing all origins while allow_credentials is enabled poses a significant security risk.
example: 
```python 
origins = [
    "http://example.com",
    "http://anotherdomain.com",
]
app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
   
]
```
 but it's quite common to set allow_origins = ["*"] during the development phase to simplify testing and development. This configuration allows requests from any origin

**Reply by Imanflow** [reply]
I've set it to True, but the issue of a mysterious `access-control-allow-origin: *` appearing out of thin air for OPTIONS method persists

**Comment by Imanflow** [comment]
Issue solved, apparently the server had a proxy configuration.
